platform/ios/src/MGLMapView.mm

```diff
@@ -5353,9 +5353,13 @@ - (void)insertCustomStyleLayerWithIdentifier:(NSString *)identifier preparationH
 {
     NSAssert(identifier, @"Style layer needs an identifier");
     MGLCustomStyleLayerHandlers *context = new MGLCustomStyleLayerHandlers(preparation, drawing, completion);
-    _mbglMap->addLayer(std::make_unique<mbgl::style::CustomLayer>(identifier.UTF8String, MGLPrepareCustomStyleLayer,
-                                                                  MGLDrawCustomStyleLayer, MGLFinishCustomStyleLayer, context),
-                       otherIdentifier ? mbgl::optional<std::string>(otherIdentifier.UTF8String) : mbgl::optional<std::string>());
+    try {
+        _mbglMap->addLayer(std::make_unique<mbgl::style::CustomLayer>(identifier.UTF8String, MGLPrepareCustomStyleLayer,
+                                                                      MGLDrawCustomStyleLayer, MGLFinishCustomStyleLayer, context),
+                           otherIdentifier ? mbgl::optional<std::string>(otherIdentifier.UTF8String) : mbgl::optional<std::string>());
+    } catch (std::runtime_error & err) {
+        [NSException raise:@"MGLRedundantLayerIdentiferException" format:@"%s", err.what()];
```

Comment:
I just realized we’ve been saying “identifer” instead of “identifier”, both for layers and sources. I can fix that typo in a separate PR, since we also need to add a note to the documentation for the affected methods about `MGLRedundantLayerIdentifierException` ([prior art involving sources](https://github.com/mapbox/mapbox-gl-native/blob/93b5ea060237de22b347527f3dc95e6f7cdb7251/platform/darwin/src/MGLStyle.h#L204-L206)).